app/emoji.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"bytes"
+	"image"
+	"image/color/palette"
+	"image/draw"
+	"image/gif"
+	_ "image/jpeg"
+	"image/png"
+	"io"
+	"mime/multipart"
+
+	"github.com/disintegration/imaging"
+	"github.com/mattermost/platform/model"
+)
+
+const (
+	MaxEmojiFileSize = 1000 * 1024 // 1 MB
+	MaxEmojiWidth    = 128
+	MaxEmojiHeight   = 128
+)
+
+func UploadEmojiImage(id string, imageData *multipart.FileHeader) *model.AppError {
```

Comment:
You refactored these out of the `api` package right? I know we're using these new functions with `api4` but can we use them with `api` as well? Just so we don't have duplicate code anywhere